### PR TITLE
ListBucket pagination

### PR DESCRIPTION
**Pull request by jsantos17**
This is essentially a rewrite of `children`.

### CODE REVIEW

**Comment by jsantos17** [comment]
[ch139]

**Comment by jsantos17** [comment]
I have a non-recursive implementation using `Stream.iterEval` and `takeWhile` too, but turns out that's a lot more complex than this recursive implementation.  I still haven't figured out how to write integration test this in Travis, but I have been able to do some local tests and this works well. 

Will write some unit tests and then assign this to someone for merge after that.

**Comment by jsantos17** [comment]
[WIP] while I figure out Travis problems.

File: datasource/src/main/scala/quasar/physical/s3/impl/children.scala [hunk 1]
```diff
@@ -54,23 +64,66 @@ object children {
...
     bucket: Uri,
     dir: APath,
-    sign: Request[F] => F[Request[F]]): F[Option[List[PathSegment]]] = {
-    val FO = Functor[F].compose[Option]
+    sign: Request[F] => F[Request[F]])
+      : F[Option[Stream[F, PathSegment]]] = {
+    val msg = "Unexpected failure when streaming a multi-page response for ListBuckets"
+    val stream0 =
+      handleS3(fetchResults(client, bucket, dir, None, sign)) map (results =>
+        Stream.iterateEval(results) {
```
Comment (wemrysi) [inline datasource/src/main/scala/quasar/physical/s3/impl/children.scala]: I think `Stream.unfoldEval` is a better fit here.

**Pull request merged**